client/src/core-components/autocomplete-dropdown.js

```diff
@@ -0,0 +1,280 @@
+
+        event.preventDefault();
+
+        this.setState({
+            itemsSelected: newList,
+            opened: false,
+            highlightedIndex: 0,
+        });
+
+        onChange && onChange(newList);
+        onRemoveClick && onRemoveClick(itemId);
+        this.searchApi("", newList);
+    }
+
+    onChangeDropDown(e){
+        const {
+            onChange,
+            onTagSelected,
+        } = this.props;
+
+        if (this.getDropdownList().length) {
+            const itemSelected = this.getDropdownList()[e.index];
+            const newList = [...this.getValue(), itemSelected];
+
+            this.setState({
+                itemsSelected: newList,
+                inputValue: "",
+                highlightedIndex: 0,
+            });
+
+            onChange && onChange(newList);
+            onTagSelected && onTagSelected(itemSelected.id);
+            this.searchApi("", newList);
+    
+        }
+
+    }
+
+    onChangeInput(str){
+        const { getItemListFromQuery, } = this.props;
+
+        this.setState({
+            inputValue: str,
+            opened: true,
+            highlightedIndex: 0,
+        });
+
+        if (getItemListFromQuery !== undefined) {
+            this.setState({
+                loading: true,
+            });
+
+            this.setTimeout(str);
+
+        }
+
+    }
+
+    onMenuToggle(b){
+        this.setState({
+            opened: b,
+        });
+    }
+    
+    onHighlightedIndexChange(n){
+        this.setState({
+            highlightedIndex: n,
+        });
+    }
+
+    onKeyDown(event){
+        const {
+            onChange,
+            onRemoveClick,
+        } = this.props;
+
+        if(this.props.disabled) {
+            event.stopPropagation();
+            event.preventDefault();
+
+            return;
+
+        }
+
+        if (keyCode(event) === "space"){
+            event.stopPropagation();
+
+        }
+
+        if (keyCode(event) === "backspace" && this.state.inputValue === ""){
+            const newList = this.getValue().slice(0,this.getValue().length-1);
+            this.setState({
+                itemsSelected: newList,
+                highlightedIndex: 0,
+            });
+
+            onChange && onChange(newList);
+
+            if (this.getValue().length) {
+                const itemId = this.getValue()[this.getValue().length-1].id;
+
+                onRemoveClick && onRemoveClick(itemId);
+            }
+
+            this.searchApi("", newList);
+
+        }
+
+    }
+
+    searchApi(query, blacklist=this.getValue()) {
+        const { getItemListFromQuery, } = this.props;
+
+        if (getItemListFromQuery !== undefined) {
+
+            getItemListFromQuery(query, blacklist.map(item => item.id))
+            .then(res => {
+                this.setState({
+                    items2: res,
+                    loading: false,
+                });
+            })
+            .catch(() => {
```

Comment:
you should to connect the .catch with the last parenthesis of the .then.All in the same line